### PR TITLE
more tests

### PR DESCRIPTION
**Pull request by tzemanovic**
## Describe your changes

related to #1497

Depends-On: #3697

- add couple basic tests in core
- removes "integration" feature

## Checklist before merging 
- [ ] If this PR has some consensus breaking changes, I added the corresponding `breaking::` labels
    - This will require 2 reviewers to approve the changes

### CODE REVIEW

**Comment by tzemanovic** [comment]
I rm'd the "integration" feature in [9dc5a13](https://github.com/anoma/namada/pull/3704/commits/9dc5a1391bca5e39138a4d7dcf0def05adaa5667), but in CI it's still running with `--feature integration` from main so this is expected to fail

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -607,4 +612,37 @@ jobs:
...
       - name: Stop sccache
         if: always() && steps.sccache.conclusion == 'success'
-        run: sccache --stop-server || true
\ No newline at end of file
+        run: sccache --stop-server || true
+
+  upload-coverage:
+    runs-on: [self-hosted, 8vcpu-16ram-ubuntu22-namada-x86]
+    container: 
+      image: ghcr.io/heliaxdev/namada-ci:namada-main
```
Comment (Fraccaman) [inline .github/workflows/ci.yml:620]: lets not use a custom runner for this, we can use github free runners

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -607,4 +612,37 @@ jobs:
...
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+        if: ${{ github.event_name != 'pull_request_target' }}
+      - name: Checkout PR
+        uses: actions/checkout@v4
+        if: ${{ github.event_name == 'pull_request_target' }}
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
```
Comment (Fraccaman) [inline .github/workflows/ci.yml:632]: we dont need to checkout the repo, just download the artifacts and upload

File: .github/workflows/ci.yml (hunk 1 above)
Comment (tzemanovic) [reply .github/workflows/ci.yml]: [542992b](https://github.com/anoma/namada/pull/3704/commits/542992b5193d3a66d410c24a6f23abb0b33a3840)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (tzemanovic) [reply .github/workflows/ci.yml]: [542992b](https://github.com/anoma/namada/pull/3704/commits/542992b5193d3a66d410c24a6f23abb0b33a3840)

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -615,21 +615,13 @@ jobs:
         run: sccache --stop-server || true
 
   upload-coverage:
-    runs-on: [self-hosted, 8vcpu-16ram-ubuntu22-namada-x86]
+    runs-on: [ubuntu-latest]
     container:
```
Comment (Fraccaman) [inline .github/workflows/ci.yml:619]: oh sorry we can remove also the `container` thingy

**Review by Fraccaman — APPROVED** [review]

**Pull request merged**